### ISSUE TITLE
Need to support conv3d conversion

### ISSUE THREAD

**Issue by seockho-kim** [Samsung/TICO#430]
## What

Some VL models(ex, qwen-vl) use conv3d in their vision encoder,
so tico needs a pass to convert conv3d to other operations, like conv2d

**Comment by mhs4670go** [comment]
Seems that the current implementation correctly decomposes `Conv3d` into a sum of kT independent `Conv2d` ops along the temporal axis. This is mathematically correct in the general case and works for arbitrary stride/dilation/padding.

However, for Qwen2.5-VL’s patch embedding `Conv3d`, I think we can use a simpler and strictly equivalent transformation that is both smaller in graph size and more efficient, under specific (but common) conditions.

We can concatenate the kT temporal frames into the channel dimension and reshape the `Conv3d` weight accordingly. Then, we can run a single `Conv2d`.

This single-Conv2d replacement is exact only if:

- kernel_T == stride_T
- padding_T == 0
- dilation_T == 1
- no temporal overlap

These conditions hold for Qwen2.5-VL patch embedding, but not for general `Conv3d` layers. So, we can keep the current your  implementation as the general `Conv3d` fallback.

I'll implement this logic soon. Please feel free to give your opinions.

```python
import torch
import torch.nn.functional as F

def demo(device="cpu", dtype=torch.float32):
    torch.manual_seed(0)

    N, C_in, C_out = 1, 3, 1280
    T, H, W = 6, 28, 28
    kT, kH, kW = 2, 14, 14
    sT, sH, sW = 2, 14, 14

    x = torch.randn(N, C_in, T, H, W, device=device, dtype=dtype)
    w3 = torch.randn(C_out, C_in, kT, kH, kW, device=device, dtype=dtype)

    y3 = F.conv3d(x, w3, bias=None, stride=(sT, sH, sW), padding=0, dilation=1, groups=1)

    T_out = (T - kT) // sT + 1
    x_grp = x.reshape(N, C_in, T_out, kT, H, W)                 # (N,C,T_out,2,H,W)
    x2 = x_grp.permute(0, 2, 1, 3, 4, 5).contiguous()           # (N,T_out,C,2,H,W)
    x2 = x2.view(N * T_out, C_in * kT, H, W)                    # (N*T_out, C*2, H, W)

    w2 = w3.contiguous().view(C_out, C_in * kT, kH, kW)         # (C_out, C*2, 14, 14)

    y2 = F.conv2d(x2, w2, bias=None, stride=(sH, sW), padding=0, dilation=1, groups=1)

    H_out = (H - kH) // sH + 1
    W_out = (W - kW) // sW + 1
    y2 = y2.view(N, T_out, C_out, H_out, W_out).permute(0, 2, 1, 3, 4).contiguous()

    max_abs = (y3 - y2).abs().max().item()
    print("max |y3-y2| =", max_abs)

demo()
```

**Comment by dayo09** [comment]
Hi. I want to update this conv3d conversion workflow.

As @mhs4670go demonstrated, Qwen3-Vl's patch embedding layer assumes that same stride == same kernel, thus can be lowered to conv2d. Not only that, it's actually `Linear` projection.

After figuring out that fact, we also assessed the impacts of both Conv3d-to-Conv2d conversion and Conv3d-to-linear conversion with @llFreetimell and @parjong.
- Converting it into `linear` may harm performance as some reshape operations could be added.
- Thus, we shortly decided to perform conv2d with (1, 1, -1, D * H * W) as the input shape comes as (Batch, D * H * W) - @llFreetimell is still inspecting the previous layer to ensure this.

[this proposal](https://github.com/Samsung/TICO/issues/475#issuecomment-3875704417) is based on above condition. I will update this to the issue.
